### PR TITLE
TabPanel: Add 40px size prop to tab Button

### PR DESCRIPTION
**Pull request by mirka**
In preparation for #65751

## What?

Adds `__next40pxDefaultSize` prop to the Button in TabPanel.

## Testing Instructions

See the TabPanel story in Storybook. There should be no visual changes.

### CODE REVIEW

File: packages/components/src/tab-panel/style.scss [hunk 1]
```diff
@@ -7,12 +7,10 @@
 	}
 }
 
-// This tab style CSS is duplicated verbatim in
-// /packages/edit-post/src/components/sidebar/settings-header/style.scss
```
Comment (mirka) [inline packages/components/src/tab-panel/style.scss:11]: This stylesheet was completely removed in #55360.

File: packages/components/src/tab-panel/index.tsx [hunk 2]
```diff
@@ -220,6 +220,7 @@ const UnforwardedTabPanel = (
 							) }-view` }
 							render={
 								<Button
+									__next40pxDefaultSize
```
Comment (tyxla) [inline packages/components/src/tab-panel/index.tsx:223]: It's a bit of a bummer that this prop is a no-op here due to the style override.

File: packages/components/src/tab-panel/style.scss [hunk 3]
```diff
@@ -7,12 +7,10 @@
...
 .components-tab-panel__tabs-item {
 	position: relative;
 	border-radius: 0;
-	height: $grid-unit-60;
+	height: $grid-unit-60 !important; // https://github.com/WordPress/gutenberg/pull/65075#discussion_r1746282734
```
Comment (tyxla) [inline packages/components/src/tab-panel/style.scss:13]: 👍 I'd love it if we can clean those up at some point.

**Review by tyxla — APPROVED** [review]
LGTM 👍

File: packages/components/src/tab-panel/style.scss (hunk 3 above)
Comment (mirka) [reply packages/components/src/tab-panel/style.scss:13]: It should be pretty simple to sweep up after the 40px becomes the default.

**Pull request merged**